ext/mysqli/mysqli_prop.c

```diff
@@ -71,38 +71,71 @@ if (!obj->ptr) { \
 	p = (MYSQL_STMT *)((MY_STMT *)((MYSQLI_RESOURCE *)(obj->ptr))->ptr)->stmt; \
 }
 
-#define MYSQLI_MAP_PROPERTY_FUNC_LONG( __func, __int_func, __get_type, __ret_type, __ret_type_sprint_mod)\
+#define MYSQLI_MAP_PROPERTY_FUNC_LONG_OR_STR( __func, __int_func, __get_type, __ret_type, __ret_type_sprint_mod)\
 static int __func(mysqli_object *obj, zval *retval, bool quiet) \
 {\
 	__ret_type l;\
 	__get_type;\
-	if (!p) {\
-		ZVAL_NULL(retval);\
-	} else {\
+	if (p) {\
```

Comment:
It's not entirely clear for me whether `p` can be `NULL`. As far as I see it's only possible if
```
p = (MYSQL *)((MY_MYSQL *)((MYSQLI_RESOURCE *)(obj->ptr))->ptr)->mysql;
```
in `MYSQLI_GET_MYSQL` is `NULL`. But it shouldn't be, right?

Reply:
Yes, I think NULL is not possible because we always require either MYSQLI_STATUS_INITIALIZED or MYSQLI_STATUS_VALID. I think it would be good to replace these if(p) checks with ZEND_ASSERT(p) to make sure we know if this is not true.